src/NServiceBus.Metrics/Probes/DurationProbe.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using NServiceBus;
+
+class DurationProbe : Probe, IDurationProbe
+{
+    public DurationProbe(string name, string description) : base(name, description)
+    {
+    }
+
+    public void Register(Action<TimeSpan> observer)
+    {
+        observers += observer;
```

Comment:
But the probe is attached to the endpoint. It's done in `MetricsFeature`. When it dies, it takes its probes with itself. I see that your examples are re-entrant in terms of, that the same process can spawn multiple endpoints of the same type. Still, I see no flaw i it.